File: entregable_profe.php
<?php
//FUNCIONES
function imprimir_cualidades($cualidades) {
    foreach ($cualidades as $cualidad) {
        echo $cualidad . "\n";
    }
}
if(!function_exists("readline")) {
    function readline($prompt = null){
        if($prompt){
            echo $prompt;
        }
        $fp = fopen("php://stdin","r");
        $line = rtrim(fgets($fp, 1024));
        return $line;
    }
}

function agregar_cualidades(&$cualidades) {
    $flag = true;
    while ($flag) {
        if (readline("Deseas agregar más condiciones?(Y/N)") == "Y") {
            $cualidades[] = readline("Escribe la propiedad: ");
        } else {
            $flag = false;
        }
    }
}

function cuestionar_cualidades($cualidades) {
    $score = 0;
    foreach ($cualidades as $cualidad) {
        $prompt = readline("Es $cualidad? (Y/N)");
        if ($prompt != 'N') {
            $score = $score + 1;
        }
    }
    return $score;
}

function calcular_score($score, $len) {
    if (($score / $len) == 1) {
        echo "Es perfecto\n";
    } elseif (($score / $len) >= 0.8) {
        echo "Apliquese o se lo ganan\n";
    } elseif (($score / $len) >= 0.6) {
        echo "Momeno\n";
    } elseif (($score / $len) >= 0.4) {
        echo "No te lo recomiendo\n";
    } elseif (($score / $len) <= 0.2) {
        echo "Corre y no mires atrás\n";
    }
}

//VARIABLES
$qualities = ["alto", "bronceado", "guapo", "rico", "sexy", "tierno", "rudo"];

//PROGRAMA
echo "Vamos a ver si la persona que piensas te conviene\n";
echo "Estas son las cualidades que actualmente validamos\n";
imprimir_cualidades($qualities);
agregar_cualidades($qualities);
echo "vamos a comenzar\n";

$score = cuestionar_cualidades($qualities);
calcular_score($score, count($qualities));
?>

File: entregable_profe.cs
using System;
using System.Collections.Generic;
					
public class Program
{
	
	static void ImprimirCualidades(List<string> cualidades)
    {
        foreach (var cualidad in cualidades)
        {
            Console.WriteLine(cualidad);
        }
    }

    static void AgregarCualidades(List<string> cualidades)
    {
        bool flag = true;
        while (flag)
        {
            Console.WriteLine("Deseas agregar más condiciones?(Y/N)");
            if (Console.ReadLine() == "Y")
            {
                Console.WriteLine("Escribe la propiedad: ");
                cualidades.Add(Console.ReadLine());
            }
            else
            {
                flag = false;
            }
        }
    }

    static int CuestionarCualidades(List<string> cualidades)
    {
        int score = 0;
        foreach (var cualidad in cualidades)
        {
            Console.WriteLine("Es {0}? (Y/N)",cualidad);
            if (Console.ReadLine() != "N")
            {
                score++;
            }
        }
        return score;
    }

    static void CalcularScore(int score, int len)
    {
        double ratio = (double)score / len;
        if (ratio == 1)
        {
            Console.WriteLine("Es perfecto");
        }
        else if (ratio >= 0.8)
        {
            Console.WriteLine("Apliquese o se lo ganan");
        }
        else if (ratio >= 0.6)
        {
            Console.WriteLine("Momeno");
        }
        else if (ratio >= 0.4)
        {
            Console.WriteLine("No te lo recomiendo");
        }
        else if (ratio <= 0.2)
        {
            Console.WriteLine("Corre y no mires atrás");
        }
    }
	
	public static void Main()
	{
		
		List<string> qualities = new List<string> { "alto", "bronceado", "guapo", "rico", "sexy", "tierno", "rudo" };

        //PROGRAMA
        Console.WriteLine("Vamos a ver si la persona que piensas te conviene");
        Console.WriteLine("Estas son las cualidades que actualmente validamos");
        ImprimirCualidades(qualities);
        AgregarCualidades(qualities);
        Console.WriteLine("Vamos a comenzar");

        int score = CuestionarCualidades(qualities);
        CalcularScore(score, qualities.Count);
	}
}